### ISSUE TITLE
WCAG 1.4.3 Violation for Dark Mode Browser on Homepage

### ISSUE THREAD

**Issue by meepsterleapster** [wordplaydev/wordplay#399]
## Expected behavior

Subtitle text under links are clearly readable in all cases and provides the user with a clear understanding of what each  link does

## Actual behavior

Dark mode contrast for subtitles fails WCAG expected contrast ration of 4:5:1. This makes subtitles hard to see on the homepage, creating an accessibility issue.

##Reproduction Steps for Manual Test

1. Go to 'Settings'
2. Click on 'Themes'
3. Select 'Any Dark Mode Theme'
4. Reload Wordplay Page
5. Scroll Down to any gray colored subtitles

_Steps for automatic reproduction coming soon_

## Screenshots
<img width="372" alt="image" src="https://github.com/wordplaydev/wordplay/assets/155786514/19241f0f-4675-4d1a-a906-4f5a9e5a7ed4">

## Environment

Desktop

-   OS + version: [Windows 22H2]
-   Browser + version [Chrome Version 121.0.6167.185 (Official Build) (64-bit)]

**Comment by meepsterleapster** [comment]
If on a device which allows for chromium extensions, a manual test can also be run with with the follow extension (enable the extension to run on the Wordplay Homepage): https://accessibilityinsights.io/downloads/

**Comment by rf-uw** [comment]
<img width="1440" alt="Screenshot 2024-02-21 at 4 27 03 PM" src="https://github.com/wordplaydev/wordplay/assets/156073685/ac1b5c29-b569-4d0c-b626-19f152e3eef5">

Same issue with headers and subheaders in the tutorial and regular code window - highlighted some examples, but the borders and some code symbols (brackets/parentheses, commas, forward slashes) use the same color so are also non-compliant.


<img width="1440" alt="Screenshot 2024-02-21 at 4 47 28 PM" src="https://github.com/wordplaydev/wordplay/assets/156073685/63a5193f-66cb-4e77-94cf-8c7e0834be97">

I'm unsure if these buttons (dark grey in the selected state, black/transparent in the unselected state) qualify as noncompliant - the background color of the buttons doesn't contrast enough with the site background but the text on the buttons does contrast enough with both the buttons and the background.

**Comment by alexhuynh0** [comment]
I am assigning myself to and commenting on this issue. I am open to other people collaborating with me on this issue.

Plans:
- Run Manual Test
- Locate parts of the implementation that are responsible for font color
- Attempt direct fixes such as manually changing the font color in dark mode
- Etc.

**Comment by amyjko** [comment]
Thanks for the update Alex. (You didn't assign yourself, so I've assigned you).

**Comment by leo3tian** [comment]
Hi, I wanted to join this issue real quick! I noticed nobody is assigned yet so please let me know if you are also working on this issue. Let's get this done as quick as possible because the lack of contrast in these colors can really affect accessibility for important functions.
### Here's what I have figured out so far:
1. The coding part of this issue is quick, it's just changing the value of **line 130 in the app.html file**
2. Light-gray subtitles seems to work best - White also works, but doesn't result in much contrast between the subtitle color and body text color. Both pass the WCAG expected contrast test.
3. The floating glyphs in the background remain super hard to see, unless they are fully white.
### Here are some screenshots:
# Current 
![image](https://github.com/wordplaydev/wordplay/assets/87582530/d67187c9-93cf-44a8-8435-871cc054bd39)
![image](https://github.com/wordplaydev/wordplay/assets/87582530/e44d1c02-9cb7-4854-a276-d68dec848c5a)
# Light Gray (hex code #868686)
![image](https://github.com/wordplaydev/wordplay/assets/87582530/c83a3873-72ee-49a4-9c73-a3b59dc404b8)
![image](https://github.com/wordplaydev/wordplay/assets/87582530/812473b8-632b-45fe-a33a-142243ee387e)

Next steps:
Any Design feedback is welcomed - is the contrast between the gray and white enough? Any color combinations that I should try?
I'll look into how this affects other parts of the app, such as the editor view and project views. I think colors for the learn page still need some work, since even this gray is too little contrast for the bits of small text.

**Comment by amyjko** [comment]
(Remember to tag us when you want design feedback: Isabel is also helping with design reviews). 

For dark mode, I don't think it's important that the subtitles be grey, since the headers are all currently links. Explore just making them specifically `--wordplay-foreground`

**Comment by leo3tian** [comment]
Gotcha, this is what it looks like with subtitles as --wordplay-foreground:
# Front Page:
![image](https://github.com/wordplaydev/wordplay/assets/87582530/06069353-2f32-4930-8cbb-ce819a397f6a)
# Project View:
![image](https://github.com/wordplaydev/wordplay/assets/87582530/d021c904-10d4-4e04-a538-1ce43901c2a6)
Everything looks good to me so I've made a [new branch](https://github.com/wordplaydev/wordplay/tree/dark-mode-subtitle-color) with the one change. Would be glad if @amyjko or @joonpiter could review!

**Comment by amyjko** [comment]
Lots of possible remedies here; make a judgement, submit a PR, and it's iterate.

**Comment by leo3tian** [comment]
As the quarter comes to a close, here's a summary of my work on this issue - so far I've tested a solution to this issue and created a pull request with the proposed changes. Here's the link to the pull request I've made for this issue: https://github.com/wordplaydev/wordplay/pull/458.
I've also identified another WCAG violation (WCAG 1.4.1 "Ensure links are distinguished from surrounding text in a way that does not rely on color") that may be added to this issue. Here are some relevant screenshots:
![image](https://github.com/wordplaydev/wordplay/assets/87582530/1ed916b3-031a-4bc1-ae45-1642c9674661)
![image](https://github.com/wordplaydev/wordplay/assets/87582530/dac5025c-3b37-43b7-a703-73be47ca24f6)
I plan on continuing work on this issue over the summer and in the coming fall quarter.

**Comment by amyjko** [comment]
Thank you @leo3tian. New issues should be new issues; please submit a new one for the new problem.

**Comment by amyjko** [comment]
No PR. Resolved by 7c7d051d61aa7cfd88793f976fa54ebf89f8ab22.
